### ISSUE TITLE
Autoplay not blocked on some sites regardless of settings configuration

### ISSUE THREAD

**Issue by Brave-Matt** [brave/brave-browser#1761]
## Description 
Autoplay works as intended for some sties whether Brave is told to ask before allow or if blocked by default:
![ignautoplay](https://user-images.githubusercontent.com/3299098/47261787-68967180-d48c-11e8-8072-8b70762c0fd2.gif)

Some websites will autoplay regardless of setting:
![youtubeautoplay](https://user-images.githubusercontent.com/3299098/47261827-0ab65980-d48d-11e8-920c-fcebe3a1bcd1.gif)

![metacafeautoplay](https://user-images.githubusercontent.com/3299098/47261861-c11a3e80-d48d-11e8-9194-2b366ee274f5.gif)

Vimeo also autoplays regardless of settings. I haven't had time to test this on further sites but it's fairly prevalent 

## Steps to Reproduce

   1.  Launch Brave (Beta, Dev), set autoplay to ask permission, block, or block a target page.
   2. Visit [one of the above or another page that streams video] page, click video link

## Actual result:

Video will autoplay and if Brave was told to ask first, permission will not be asked. 

## Expected result:
Autoplay should be blocked.

## Reproduces how often: 

Easily, but not for every website.

## Brave version (brave://version info)

Beta - v0.56.5
Dev - v0.57.1
### Reproducible on current release:
- Does it reproduce on brave-browser dev/beta builds?  Yes
- Does it reproduce on browser-laptop? No

### Website problems only:  
- Does the issue resolve itself when disabling Brave Shields? No
- Is the issue reproducible on the latest version of Chrome? No

### Additional Information

List of sites confirmed to bypass autoplay settings (will update list if I find more):

1. youtube.com
2. metacafe.com
3. Vimeo.com

**Comment by srirambv** [comment]
Autoplay block works just fine only when the correct URL is added to the block list. There are 2 ways to achieve this

First method (easy)
1. Visit https://youtube.com
2. Click on site settings (URL bar icon)
3. Click Site Settings 
4. Change Autoplay from `Block (Default)` to `Block`
5. Go back to YouTube and click any video, doesn't autoplay
![autoplay3](https://user-images.githubusercontent.com/17010094/47266835-d0d76880-d509-11e8-89ce-a98a55ee62ec.gif)

Second method (Advanced)
1. Open Settings-> Privacy -> Content Settings -> Autoplay
2. Click `Add` under block
3. Type `https://www.youtube.com:443` and click ok
4. Open YouTube and play any video, its blocked by default
![autoplay2](https://user-images.githubusercontent.com/17010094/47266883-6bd04280-d50a-11e8-8c27-faedd163c70c.gif)

Note: for second method if you just add `https://youtube.com` or `youtube.com` manually to the list it won't work. You need to add the FDN with `:443` for it to be blocked. 

The `Block (Default)` value is overridden by the autoplay whitelist so it has to be changed to `Block` for it to stop playing videos when page is loaded.  I don't believe this is an issue with autoplay. @Brave-Matt we should probably have a support page created for this. 

@bbondy @darkdh @bridiver we should probably fix the way user manually adds the site name to the block list. It only works if you add `https://www.` and `:443`. Should probably respect and block when only the URL name is entered or should auto add these things when user enters an url

**Comment by bbondy** [comment]
The following sites are intentionally allowed by default because the sole purpose of the site is to watch videos:
https://github.com/brave/brave-core/blob/d62d279f8d5bf6d3bbea648d5291d7e4316834b3/renderer/brave_content_settings_observer.cc#L243

Other sites are block by default. 

You can override either default by adding a content setting to block or allow. 

Knowing that, is there still an issue here?

**Comment by srirambv** [comment]
This is an invalid issue, its working as intended as far as I can tell. We need a support document on that to be created on how to block autoplay on sites which are in the whitelist using method 1 from above.

The only issue I see is manually entering a site name doesn't block it requires `https://www.` prefix and the port number `:443` added. Only then the block functionality works for manually added sites.

**Comment by darkdh** [comment]
This is how chromium content settings rule work, not autuplay specific.
If you are not familiar with the rules, just change content settings per site by UI.

**Comment by bbondy** [comment]
I think we can modify how we match content setting rules to allow subdomain and port specifications.  If that still remains let's close this out and post a new issue for that aspect. Thanks!

**Comment by srirambv** [comment]
Closing this in favour of #1775. Need to create a support document for how to disable autoplay on whitelisted sites
